src/Xamarin.Forms.GoogleMaps.Clustering.Android/ClusterRenderer.cs

```diff
@@ -96,6 +161,72 @@ private void AddToIconCache(ICluster cluster, NativeBitmapDescriptor icon)
                 disabledBucketsCache.Add(clusterText, icon);
         }
 
+        /// <summary>
+        /// The GetClusterText
+        /// </summary>
+        /// <param name="cluster">The cluster<see cref="ICluster"/></param>
+        /// <returns>The <see cref="string"/></returns>
+        private string GetClusterText(ICluster cluster)
+        {
+            return GetClusterText(cluster.Size);
+        }
+
+        /// <summary>
+        /// The BucketIndexForSize
+        /// </summary>
+        /// <param name="size">The size<see cref="int"/></param>
+        /// <returns>The <see cref="int"/></returns>
+        private int BucketIndexForSize(int size)
+        {
+            uint index = 0;
+            var buckets = map.ClusterOptions.Buckets;
+
+            while (index + 1 < buckets.Length && buckets[index + 1] <= size)
+                ++index;
+
+            return (int)index;
+        }
+
+        #endregion
+
+        /// <summary>
+        /// The OnBeforeClusterRendered
+        /// </summary>
+        /// <param name="cluster">The cluster<see cref="ICluster"/></param>
+        /// <param name="options">The options<see cref="MarkerOptions"/></param>
+        protected override void OnBeforeClusterRendered(ICluster cluster, MarkerOptions options)
+        {
+            NativeBitmapDescriptor icon;
+            if (map.ClusterOptions.RendererCallback != null)
+            {
+                var descriptorFromCallback =
+                    map.ClusterOptions.RendererCallback(map.ClusterOptions.EnableBuckets ?
+                        GetClusterText(cluster) : cluster.Size.ToString());
+                icon = GetIcon(cluster, descriptorFromCallback);
+                options.SetIcon(icon);
+            }
+            else if (map.ClusterOptions.RendererImage != null)
+            {
+                icon = GetIcon(cluster, map.ClusterOptions.RendererImage);
+                options.SetIcon(icon);
+            }
+            else if (!map.ClusterOptions.EnableBuckets)
+            {
+                markerView.SetBackgroundResource(Resource.Drawable.marker_cluster);
```

Comment:
do we need custom views for that purpose? I'd try to do it using clustering library options. Have you tried it?

Reply:
Nvm it makes sense here :)